spec/que/views/que_jobs_ext_spec.rb

```diff
@@ -0,0 +1,200 @@
+
+        job = jobs_dataset.first
+        ext_job = jobs_ext_dataset.first
+        assert_equal ext_job[:queue], job[:queue]
+        assert_equal ext_job[:priority], job[:priority]
+        assert_equal ext_job[:run_at], job[:run_at]
+        assert_equal ext_job[:first_run_at], job[:first_run_at]
+        assert_equal ext_job[:job_class], job[:job_class]
+        assert_equal ext_job[:args], job[:args]
+        assert_equal ext_job[:job_schema_version], job[:job_schema_version]   
+
+        jobs_dataset.delete
+
+        assert_equal 0, jobs_dataset.count
+        assert_equal 0, jobs_ext_dataset.count
+      end
+
+      it "should include additional lock data" do
+        locker
+
+        sleep_until_equal(1) { DB[:que_lockers].count }
+
+        TestJob.enqueue
+
+        
+        assert_equal 1, jobs_dataset.count
+        assert_equal 1, jobs_ext_dataset.count
+
+        sleep_until { active_jobs_dataset.count.positive? }
+        
+        ext_job = jobs_ext_dataset.first
+
+        assert_equal false, ext_job[:lock_id].nil?
+
+        assert_equal ext_job[:lock_id], locked_ids.first
+        assert_equal ext_job[:que_locker_pid], locked_pids.first
+
+        locker.stop!
+      end
+
+      it "should add additional updated_at" do
+        locker
+
+        sleep_until_equal(1) { DB[:que_lockers].count }
+
+        TestJob.enqueue
+
+        ext_job = jobs_ext_dataset.first
+
+        assert_equal ext_job[:run_at], ext_job[:updated_at]
+
+        sleep_until { locked_ids.count.positive? }
+        sleep_until { locked_ids.count.zero? }
+
+        ext_job = jobs_ext_dataset.first
+
+        assert_equal ext_job[:finished_at], ext_job[:updated_at]
+
+        locker.stop!
+      end
+
+      describe "should include additional status" do
+        let(:notified_errors) { [] }
+
+        it "should set status to scheduled when run_at is in the future" do
+          TestJob.enqueue(job_options: { run_at: Time.now + 1 })
+
+          assert_equal jobs_ext_dataset.first[:status], 'scheduled'
+        end
+
+        it "should set status to queued when run_at is in the past and the job is not currently running, completed, failed or errored" do
+          TestJob.enqueue(job_options: { run_at: Time.now - 1 })
+
+          assert_equal jobs_ext_dataset.first[:status], 'queued'
+        end
+
+        it "should set status to running when the job has a lock associated with it" do
+          locker
+          
+          sleep_until_equal(1) { DB[:que_lockers].count }
+          
+          TestJob.enqueue
+
+          sleep_until { locked_ids.count.positive? }
+          assert_equal jobs_ext_dataset.first[:status], 'running'
+
+          locker.stop!
+        end
+
+        it "should set status to complete when finished_at is present" do
+          locker
+          
+          sleep_until_equal(1) { DB[:que_lockers].count }
+          
+          TestJob.enqueue
+
+          sleep_until { locked_ids.count.positive? }
```

Comment:
I think there's a race condition here (and in the other tests) - in the case that this doesn't run until the job's already completed, the tests are going to get stuck / timeout. I know you've added a sleep to the job's run method (which is a separate issue), but it could still happen under high system load